packages/react-checkbox/src/components/Checkbox/useCheckbox.tsx

```diff
@@ -78,7 +78,6 @@ export const useCheckbox = (props: CheckboxProps, ref: React.Ref<HTMLInputElemen
     }
   }
 
-  const onChange = state.input.onChange as CheckboxProps['onChange'];
```

Comment:
This should be left as-is. With your change it won't work as expected if user puts an onChange listener on the input slot, e.g. 
```jsx
<Checkbox input={{ onChange: () => /*...*/ }} />
```
That's not recommended, but it is allowed. And with your change we'd be overwriting the value from the slot.

Reply:
I don't think that should be allowed. Otherwise the typing for the top-level `onChange` prop isn't enforced due to the overlap with native `onChange`.

Reply:
Yeah it's not ideal that it's allowed. I wonder if the primary slot prop should accept _only_ the `className` and `style` props? That would eliminate the ability to specify the same prop in two places, and also fix this problem when native props have special handling by the component.